content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -0,0 +1,183 @@
+---
+title: 'Create custom Prisma Client queries'
+metaTitle: 'Prisma Client extensions: query component (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+tocDepth: 3
+---
+
+<TopBlock>
+
+You can use the `query` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component type to hook into the query life-cycle and modify an incoming query or its result. We introduced this feature in version 4.7.0.
+
+You can use the `query` extension component as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases). Unlike middlewares, the `query` extension component gives you end-to-end type safety. You can also use it to provide [user isolation](/concepts/components/prisma-client/client-extensions#extended-clients) in a row-level security (RLS) scenario. [Learn more](#query-extensions-versus-middlewares).
+
+</TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create and modify queries
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an [extended client](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions). An extended client is a variant of the standard Prisma Client that is wrapped by one or more extensions.
+
+Use the `query` extension component to create and modify queries. You can add or modify a custom query in the following:
+
+- [A specific operation in a specific model of your schema](#add-or-modify-a-custom-query-for-a-specific-operation-in-a-specific-model)
```

Comment:
I wonder if it's necessary to mention "model of your schema", "in a specific model" sounds easier

Reply:
Yep, I agree - I'll change the instances of this